### PR TITLE
feat: add support for OtelWriter

### PR DESCRIPTION
**Pull request by Ravio1i**
This PR adds the capability of an OtelWriter to Pollect. This will allow instead of exposing the metrics at a local port to send them to a otel collector.

### Otel Collector

You can run a otel collector + Prometheus locally to test this setup

```yaml
# otel-config.yml
receivers:
  otlp:
    protocols:
      http:
        endpoint: "0.0.0.0:4318"
  
exporters:
  prometheus:
    endpoint: "0.0.0.0:8899"

service:
  pipelines:
    metrics:
      receivers: [otlp]
      processors: [batch]
      exporters: [prometheus]
```

Run otel collector locally in a container

```bash
docker run --rm --name otel-collector -p 4318:4318 -p 8899:8899 -v $(pwd)/otel-collector-config.yml:/otel-config.yml otel/opentelemetry-collector:latest --config /otel-config.yml
```

### Prometheus

```yaml
# prometheus.yml
global:
  scrape_interval: 10s

scrape_configs:
  - job_name: 'otel-collector'
    scrape_interval: 5s
    static_configs:
      - targets: ['otel-collector:8899']
    scrape_timeout: 1s
```

Run prometheus locally in container

```yaml
docker run --rm \                                                                                                                                                                             
  --name prometheus \
  -p 9090:9090 \
  -v $(pwd)/prometheus.yml:/etc/prometheus/prometheus.yml \
  prom/prometheus
```

### Pollect

Running with simple `config.yml`

```yaml
---
tickTime: 1
threads: 4
writer:
  type: Otel
executors:
  - collection: pollect
    sources:
      - type: LoadAvg
```

```bash
pollect --config config.yml
```

image

### CODE REVIEW

File: pollect/writers/OtelWriter.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        There is no inbuilt way to obtain a gauge from the meter, so we store them in a dictionary
+        """
+        if name in self.gauges.keys():
+            return self.gauges[name]
+        
+        gauge = self.meter.create_gauge(name=name)
+        self.gauges[name] = gauge
+        return gauge
+    
+    def get_attributes_from_labels(self, labels: list, label_values: list):
```
Comment (davidgiga1993) [inline pollect/writers/OtelWriter.py:54]: Please specify the exact list type and return value type

File: pollect/writers/OtelWriter.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    def start(self):
+        pass
+
+    def stop(self):
+        pass
+
+    def get_metric_name(self, value_set: ValueSet, value_obj: Value):
+        """
+        Converts the value set and value object to a metric name
+        Replaces all invalid characters with underscores
```
Comment (davidgiga1993) [inline pollect/writers/OtelWriter.py:34]: Why was this required? Afaik otel allows any utf8 chars as metric names. Do you have a spec for this?

Comment (Ravio1i) [reply pollect/writers/OtelWriter.py:34]: It may not necessarily be required, it was basically a take over from `PrometheusWriter`

https://github.com/davidgiga1993/pollect/blob/0f80954c53435427de0b4c9a56685a927df06c94/pollect/writers/PrometheusWriter.py#L181-L186

Will test if its needed

Comment (Ravio1i) [reply pollect/writers/OtelWriter.py:34]: okay it can be removed safely

Even when using otel collector + prometheus

```yaml
writer:
  type: Otel
executors:
  - collection: pollect.local
    sources:
      - type: LoadAvg
```

The prometheus itself will do the translation of `pollect.local_loadavg` to `pollect_local_loadavg`

image

File: setup.py [hunk 3]
```diff
@@ -36,7 +36,9 @@ def get_version(rel_path):
         'schedule',
         'prometheus-client',
         'PyYAML',
-        'requests'
+        'requests',
+        'opentelemetry-sdk',
```
Comment (davidgiga1993) [inline setup.py:40]: I'll remove this from the actual release as pollect is shipped with as little dependencies as possible, and the most common use case is still prometheus atm.
